packages/embed-widget/src/App.tsx

```diff
@@ -115,34 +120,57 @@ function App(): JSX.Element {
             )
           );
 
-          log.debug(`Loading widget definition for ${name}...`);
+          if (shared != null) {
+            log.debug(`Loading shared object for ${name}...`);
 
-          const newDefinition = await fetchVariableDefinition(connection, name);
+            const obj = await connection.getSharedObject(name, shared);
```

Comment:
We'll need the types published to make use of this...
One thing you can do so that this doesn't break is create our own type predicate that would do this (like we did with `isPartitionedTableWithBaseTable`), e.g.

```typescript
type ConnectionWithGetSharedObject = dh.IdeConnection & {
  getSharedObject(name: string, ticket: string): Promise<unknown>;
}

function isConnectionWithGetSharedObject(connection: dh.IdeConnection): connection is ConnectionWithGetSharedObject {
  return (
    'getSharedObject' in connection &&
    typeof connection.getSharedObject === 'function'
  );
}
```

Then you can do a check before calling it, or throw if that type doesn't exist.